src/state/validation.py

```diff
@@ -68,7 +68,6 @@ def wrapper(instance: C, *args: typing.Any) -> None:
                     event: ops.EventBase
                     event, *_ = args
                     event.defer()
-                logger.exception(str(exc))
```

Comment:
Since the exception message is also reflected in the status message below I thought this is not really needed.

Reply:
I think it can be worthwhile to have the stack trace because you are catching a lot of exceptions and the stack trace provides a deeper analysis. Afaik the message is also truncated in the `juju status` output.

https://github.com/canonical/github-runner-operator/blob/b24f201fa10dddb5f430684b3e04befe27a154d3/src/charm.py#L89-L129 is how its done in the github runner charm

Reply:
got it, I added back the exception log, but I'm keeping the message generic as they are all raised during initialization of charm state.